pandas/core/internals/managers.py

```diff
@@ -983,11 +983,18 @@ def fast_xs(self, loc: int) -> SingleBlockManager:
             )
             result = ensure_wrapped_if_datetimelike(result)
 
-        for blk in self.blocks:
-            # Such assignment may incorrectly coerce NaT to None
-            # result[blk.mgr_locs] = blk._slice((slice(None), loc))
-            for i, rl in enumerate(blk.mgr_locs):
-                result[rl] = blk.iget((i, loc))
+        try:
+            for blk in self.blocks:
+                # Such assignment may incorrectly coerce NaT to None
+                # result[blk.mgr_locs] = blk._slice((slice(None), loc))
+                for i, rl in enumerate(blk.mgr_locs):
+                    result[rl] = blk.iget((i, loc))
+        except TypeError:
+            if isinstance(dtype, ExtensionDtype) and not immutable_ea:
+                values = [v[loc] for v in self.arrays]
```

Comment:
this will only work in some fairly specific all-EA cases

Reply:
im skeptical this is the way to go, will discuss in the #54445 to try to get an idea of where the underlying problem is

Reply:
Maybe the solution is to use a (np.nan+0j) as the na_value, and maybe na_values should be the largest thing an array should be required to hold, not the smallest or simplest?